code/modules/mob/living/inhand_holder.dm

```diff
@@ -0,0 +1,123 @@
+//Generic system for picking up mobs.
+//Currently works for head and hands.
+/obj/item/clothing/head/mob_holder
+	name = "bugged mob"
+	desc = "Yell at coderbrush."
+	icon = null
+	icon_state = ""
+	var/mob/living/held_mob
+	var/can_head = FALSE
+	w_class = WEIGHT_CLASS_BULKY
+
+/obj/item/clothing/head/mob_holder/Initialize(mapload, mob/living/M, _worn_state, alt_worn, lh_icon, rh_icon, _can_head_override = FALSE)
+	. = ..()
+
+	if(M)
+		M.setDir(SOUTH)
+		held_mob = M
+		M.forceMove(src)
+		appearance = M.appearance
+		name = M.name
+		desc = M.desc
+
+	if(_can_head_override)
+		can_head = _can_head_override
+	if(alt_worn)
+		alternate_worn_icon = alt_worn
+	if(_worn_state)
+		item_state = _worn_state
+		icon_state = _worn_state
+	if(lh_icon)
+		lefthand_file = lh_icon
+	if(rh_icon)
+		righthand_file = rh_icon
+	if(!can_head)
+		slot_flags = NONE
+
+/obj/item/clothing/head/mob_holder/Destroy()
+	if(held_mob)
+		release()
+	return ..()
+
+/obj/item/clothing/head/mob_holder/dropped()
+	..()
+	if(isturf(loc))//don't release on soft-drops
+		release()
+
+/obj/item/clothing/head/mob_holder/proc/release(del_on_release = TRUE)
```

Comment:
What's the point of the argument? Seems like you'd never want to have a holder without a thing to hold.